storagemarket/impl/provider.go

```diff
@@ -614,13 +614,6 @@ func (p *Provider) runMigrations(ctx context.Context) ([]storagemarket.MinerDeal
 		return nil, xerrors.Errorf("failed to fetch deals during startup: %w", err)
 	}
 
-	// re-track all deals for whom we still have a local blockstore.
-	for _, d := range deals {
-		if _, err := os.Stat(d.InboundCAR); err == nil && d.Ref != nil {
-			_, _ = p.stores.GetOrOpen(d.ProposalCid.String(), d.InboundCAR, d.Ref.Root)
-		}
-	}
-
```

Comment:
This piece of code was causing the problem - we shouldn't need to do this as stores.GetOrOpen() will be called lazily